python/pyarrow/_compute.pyx

```diff
@@ -473,7 +481,7 @@ cdef class MetaFunction(Function):
 
 cdef _pack_compute_args(object values, vector[CDatum]* out):
     for val in values:
-        if isinstance(val, (list, np.ndarray)):
+        if isinstance(val, SUPPORTED_INPUT_ARR_TYPES):
             val = lib.asarray(val)
```

Comment:
Unrelated to this PR, but at some point we should accept any [ArrowArrayExportable](https://arrow.apache.org/docs/format/CDataInterface/PyCapsuleInterface.html#arrowarray-export)? @jorisvandenbossche

Reply:
We should do that indeed! 

(https://github.com/apache/arrow/issues/43410 is a general issue about accepting such objects in more places)